### ISSUE TITLE
Segfault with FractalFBm on Windows

### ISSUE THREAD

**Issue by Aurailus** [Auburn/FastNoise2#69]
Hi! I'm encountering a strange issue with FastNoise2 on windows. When trying to call GenSingle3D on a FractalFBb module, a segfault occurs. This behaviour is present in the latest commit as well as the last release, and we managed to replicate it with just the following code, which should segfault at the GenSingle3D call:

```CPP
int main(int argc, char* argv[]) {
    auto biomePerlin = FastNoise::New<FastNoise::Simplex>();
    
    auto biomeScale = FastNoise::New<FastNoise::DomainScale>();
    biomeScale->SetSource(biomePerlin);
    biomeScale->SetScale(1/1000.f);
    
    auto biomeFractal = FastNoise::New<FastNoise::FractalFBm>();
    biomeFractal->SetSource(biomePerlin);
    
    float val = biomeFractal->GenSingle3D(0, 0,0, 1337);
    
    std::cout << val << std::endl;
}```

**Comment by Auburn** [comment]
I have not tested MinGW on Windows, only MSVC and ClangCL. The code above works on latter 2 compilers.

Can you give any more details on the crash?

**Comment by SamCZ** [comment]
![image](https://user-images.githubusercontent.com/5303066/132074044-88de0cd7-7bf4-402f-bbc8-f0eacde48d47.png)

**Comment by SamCZ** [comment]
This will happen even on clean clone of repo
![image](https://user-images.githubusercontent.com/5303066/132074571-7de54b91-a7c2-4b23-8aef-45a8e30bc636.png)

**Comment by Auburn** [comment]
Well that doesn't look like a simple fix, I'll put MinGW support on my ToDo list but I'm busy with a few other tasks currently

**Comment by SamCZ** [comment]
ok, I will look into it, my discord is: `Sam Freki#9144`

**Comment by Auburn** [comment]
There is a [FastNoise Discord](https://discord.gg/SHVaVfV)

Another option is to compile it as a DLL using one of the supported compilers and use it via the C interface, but that's not an ideal solution

**Comment by SamCZ** [comment]
Yeah and its weird that the segfault is only present on some generators

**Comment by Auburn** [comment]
I just tried to test this, I didn't seem to get the segfault. Not sure if it's been fixed? The bug report linked above is still open. 

I'm testing on GNU 13.1.0

**Comment by PaoDerDoktor** [comment]
Hi ! Just to confirm, this still segfaults on master branch (using MinGw with GNU 13.2.0)  :)

> This will happen even on clean clone of repo ![image](https://user-images.githubusercontent.com/5303066/132074571-7de54b91-a7c2-4b23-8aef-45a8e30bc636.png)

**Comment by Auburn** [comment]
Could you test the NewFastSIMD branch, I may have fixed it on there

**Comment by IlaPoly** [comment]
For #142 I was using NewFastSIMD so it didn't fix it for me.

**Comment by Auburn** [comment]
Are you sure, the arguments in GenUniform3D suggest you are using the master branch

**Comment by IlaPoly** [comment]
Yes, I just rebuild the library from NewFastSIMD branch and the GenUniform3D still takes the same argument.

**Comment by DridriLaBastos** [comment]
Hi, I tested with a simple program

```c++
int main(void)
{
	float* noise = new float[10*10];
	auto Node = FastNoise::New<FastNoise::Perlin>();
	auto Output = Node->GenUniformGrid2D(noise,0,0,10,10,1,31648520);
	delete noise;
	return EXIT_SUCCESS;
}
```

my gcc version is the following :
```
gcc.exe (MinGW-W64 x86_64-ucrt-posix-seh, built by Brecht Sanders, r2) 14.2.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO   
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

and I can confirm that if I still have the bug with the default branch, I don't have it using the current last commit of the NewFastSIMD branch : 16dd2effc402641fa4503c5b7b20582372d96ed7
